native/source/mixpanel/detail/worker.cpp

```diff
@@ -135,7 +137,13 @@ namespace mixpanel
             nanowww::Response response;
             if (client.send_request(request, &response))
             {
-                mixpanel->log(Mixpanel::LogEntry::LL_TRACE, "response: " + response.content());
+                // Lock on updating the network requests allowed time
+                {
+                    std::lock_guard<std::mutex> lock(mutex);
+                    // Prevent requests until the back off time has passed
+                    network_requests_allowed_time = parse_www_retry_after(response);
```

Comment:
I am updating `failure_count` in `parse_www_retry_after`. How should I handle locking there? It seems to me if I lock again inside the function I'll be deadlocked.